### PR TITLE
[improvement](mtmv) Improve the performance for query rewritting by materialized view

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

- Limit the number of times for the query rewritting to the group
- Remove the unnecessary log and explain detail info in query

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializationContext.java [hunk 1]
```diff
@@ -150,13 +155,17 @@ public void setSuccess(boolean success) {
     /**
      * recordFailReason
      */
-    public void recordFailReason(ObjectId objectId, Pair<String, String> summaryAndReason) {
+    public void recordFailReason(StructInfo structInfo, Pair<String, String> summaryAndReason) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializationContext.java:158]: maybe use this signature is better
```suggestion
    public void recordFailReason(StructInfo structInfo, String summary, Supplier<String> reasonSupplier) {
    ...
    String reason = enableRecordFailureDetail ? reasonSupplier.get() : ""
    ...
```

Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializationContext.java]: have fixed it

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -1578,6 +1581,10 @@ public void setEnableLeftZigZag(boolean enableLeftZigZag) {
                     "whether to use a materialized view that contains the foreign table "
                             + "when using rewriting based on struct info"})
     public boolean materializedViewRewriteEnableContainExternalTable = false;
+    @VariableMgr.VarAttr(name = MATERIALIZED_VIEW_REWRITE_SUCCESS_CANDIDATE_NUM, needForward = true,
+            description = {"透明改写成功的结果集合，允许参与到CBO候选的最大数量",
```
[the remark this answers is not in the document]
Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: OK,hava fixed it

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**